Core/include/Acts/Seeding/SeedFilter.ipp

```diff
@@ -105,9 +131,99 @@ void SeedFilter<external_spacepoint_t>::filterSeeds_2SpFixed(
         continue;
       }
     }
-    outIt = std::make_pair(
-        weight, std::make_unique<const InternalSeed<external_spacepoint_t>>(
-                    bottomSP, middleSP, *topSpVec[i], zOrigin));
+
+    // increment in seed weight if number of compatible seeds is larger than
+    // numSeedIncrement
+    if (compatibleSeedR.size() > m_cfg.numSeedIncrement) {
+      weight += m_cfg.seedWeightIncrement;
+    }
+
+    int deltaSeedConf;
+    if (m_cfg.seedConfirmation) {
+      // seed confirmation cuts - keep seeds if they have specific values of
+      // impact parameter, z-origin and number of compatible seeds inside a
+      // pre-defined range that also depends on the region of the detector (i.e.
+      // forward or central region) defined by SeedConfirmationRange
+      deltaSeedConf = compatibleSeedR.size() + 1 - nTopSeedConf;
+      if (deltaSeedConf < 0 || (numQualitySeeds and !deltaSeedConf)) {
+        continue;
+      }
+      bool seedConfMinRange =
+          bottomSP.radius() < m_cfg.seedConfMinBottomRadius ||
+          std::abs(zOrigin) > m_cfg.seedConfMaxZOrigin;
+      if (seedConfMinRange and !deltaSeedConf and
+          impact > m_cfg.minImpactSeedConf) {
+        continue;
+      }
+
+      // term on the weight that depends on the value of zOrigin
+      weight += -std::abs(zOrigin) + m_cfg.compatSeedWeight;
+
+      // skip a bad quality seed if any of its constituents has a weight larger
+      // than the seed weight
+      if (weight < bottomSP.quality() and weight < middleSP.quality() and
+          weight < topSpVec[i]->quality()) {
+        continue;
+      }
+
+      if (deltaSeedConf) {
+        // if we have not yet reached our max number of quality seeds we add the
+        // new seed to outCont
+        if (numQualitySeeds < m_cfg.maxQualitySeedsPerSpMConf) {
+          // fill high quality seed
+          ++numQualitySeeds;
+          outCont.push_back(std::make_pair(
+              weight,
+              std::make_unique<const InternalSeed<external_spacepoint_t>>(
+                  bottomSP, middleSP, *topSpVec[i], zOrigin, true)));
+        } else {
+          // otherwise we check if there is a lower quality seed to remove
+          checkReplaceSeeds(bottomSP, middleSP, *topSpVec[i], zOrigin, true,
+                            weight, outCont);
+        }
+
+      } else if (weight > weightMin) {
+        // store index of lower quality seeds with a weight greater than the
+        // minimum
+        weightMin = weight;
```

Comment:
this looks like it's the maximum weight, or what am I missing?

Reply:
this is the weight of the best "lower quality" seed accepted so it is indeed not the minimum weight, should I change the name to weightMax? or something like bestWeightNoQualityConf?

Reply:
done